site/content/en/docs/tasks/dev/integrate_a_custom_job.md

```diff
@@ -85,31 +85,48 @@ Update the [Makefile](https://github.com/kubernetes-sigs/kueue/blob/main/Makefil
 
 ## Building an External Integration
 
+Building an external integration is similar to building a built-in integration, except that
+you will instantiate Kueue's `jobframework` for your CRD in a separate executable.  The only
+change you will make to the primary Kueue manager is a deployment-time configuration change
+to enable it to recognize that your CRD is being managed by a Kueue-compatible manager.
+
 Here are completed external integrations you can learn from:
    - [AppWrapper](https://github.com/project-codeflare/appwrapper)
 
 ### Registration
 
-Add your framework's GroupVersionKind to `.integrations.externalFrameworks` in [controller_manager_config.yaml](
+Configure Kueue by adding your framework's GroupVersionKind to `.integrations.externalFrameworks` in [controller_manager_config.yaml](
 https://kueue.sigs.k8s.io/docs/installation/#install-a-custom-configured-released-version).
 
-### RBAC Augmentation
+### Job Framework
 
-Kueue will need permission to get, list, and watch instances of your CRD.
+Make the following changes to an existing controller-runtime based manager for your CRD.
 
-If you are building a custom Kueue deployment, simply add a `kubebuilder:rbac` annotation to a source code file
-(for example in [integrationmanager.go](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/integrationmanager.go)) and regenerate the manifests.
+Add a dependency on Kueue to your `go.mod` so you can import `jobframework` as a go library.
 
-If you are deploying a Kueue release, modify either
-[charts/kueue/templates/rbac/role.yaml](https://github.com/kubernetes-sigs/kueue/blob/main/charts/kueue/templates/rbac/role.yaml)
-or [config/components/rbac/role.yaml](https://github.com/kubernetes-sigs/kueue/blob/main/config/components/rbac/role.yaml)
-to add the needed permissions.
+In a new `mycrd_workload_controller.go` file, implement the `GenericJob` interface, and other optional [interfaces defined by the framework](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/interface.go).
 
-### Job Framework
+Extend the existing webhook for your CRD to invoke Kueue's webhook helper methods:
```

Comment:
FYI: I'm actually changing the details about this https://github.com/kubernetes-sigs/kueue/pull/3157

Although it refers to changes that will be included in the next release, so let's go with this for now.

Reply:
yup, that's still supported. And since you keep your go types up-to-date with your CRD version, you are ok with using controller-runtime's library to build the webhook.